### ISSUE TITLE
Use standards-compliant textual representation of key derivation

### ISSUE THREAD

**Issue by abailly** [IntersectMBO/cardano-addresses#287]
In both [BIP-32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki), [BIP-44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki), and [CIP-1852](https://cips.cardano.org/cip/CIP-1852) standards, textual representation of a hierarchy of keys use some conventions:

* the prefix `m` represents the _master key_ from which other keys can be derived, 
* hardened key derivation is denoted by a single quote character following the index, eg `1852'`,
* indices are separated by forward slash character `/`. 

Full CIP-1852 compliant key derivation of a staking key would be expressed as ie. `m / 1852' / 1815' / 0' / 2 / 0`.

In the `cardano-address` CLI we use a different textual representation: The above derivation would be represented as `1852H/1815H/0H/2/0`. This representation as a few, arguably minor but still annoying problems:

* it cannot distinguish between _absolute_ and _relative_ derivations: When one writes `1H/2/3`, it's unclear whether this is a derivation from the master key or from an already derived key,
* it does not align with encoding in a programming language which would usually explicitly use a `root` or `master_key` value or object and recursively construct a derivation path from it,
* it's different from standards.

I would like to propose that we align the textual representation used in `cardano-addresses` to the one used in industry standards, while preserving backward compatibility to not break existing tools and workflows.

**Comment by abailly** [comment]
@paweljakubas @Crypto2099 @disassembler Any thoughts on this?

**Comment by paweljakubas** [comment]
I agree with the proposal to have this supported (modulo we will also support "old" style for backward compability). I am happy to handle this ticket :+1:
